src/main/java/org/passay/DigestDictionaryRule.java

```diff
@@ -0,0 +1,69 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import org.cryptacular.bean.HashBean;
+
+/**
+ * Rule for determining if a password matches a digested password that is stored in a dictionary.
+ *
+ * @author  Middleware Services
+ */
+public class DigestDictionaryRule extends AbstractDictionaryRule
+{
+
+  /** Error code for matching dictionary word. */
+  public static final String ERROR_CODE = "ILLEGAL_DIGEST_WORD";
+
+  /** Error code for matching reversed dictionary word. */
+  public static final String ERROR_CODE_REVERSED = "ILLEGAL_DIGEST_WORD_REVERSED";
+
+  /** Hash bean to use for comparing hashed passwords. */
+  private final HashBean<String> hashBean;
+
+  /** Character set to use for undigested passwords. */
+  private Charset charset = StandardCharsets.UTF_8;
+
+
+  /**
+   * Creates new digest history rule which operates on password references that were digested with the supplied hash.
+   *
+   * @param  bean  encoding hash bean
+   */
+  public DigestDictionaryRule(final HashBean<String> bean)
+  {
+    hashBean = bean;
+  }
+
+
+  /**
+   * Sets the character set to use for undigested passwords.
```

Comment:
I'd recommend a clearer explanation for this property: "Sets the character set to use when converting a candidate password to bytes prior to hashing."